google/cloud/spanner/integration_tests/backup_integration_test.cc

```diff
@@ -130,6 +137,11 @@ TEST_F(BackupTest, BackupTest) {
       << "Database " << db.database_id()
       << " not found in the backup operation list.";
 
+  // TODO(#4306) - CreateBackup() times out (rarely) log for troubleshoot
+  if (backup_future.wait_for(std::chrono::minutes(5)) != fs::ready) {
+    GCP_LOG(WARNING) << "Potential timeout in CreateBackup()" << db;
```

Comment:
I'm not sure what "potential timeout" messages are buying us.  Sometimes these operations just don't complete in the hour we've allotted.  But, OK.

Reply:
I am not sure I follow.  Are you implying that these flakes are "normal" and therefore we should ignore them?

If so, I would be glad to change the tests to say something like "this operation is taking so long that we will simply declare success on this run and move on".  The downside if that these timeouts become "common", then we would not notice.

I am actually more interested in the output from the next line, where the last N RPCs would be logged. I am hoping those will help us identify if there was something particular about the instance / database / environment that triggered the timeout.

Reply:
Ack. Closing.

Reply:
It certainly might be nice to know we were actually in a `CreateBackup()` poll loop when the timeout happened, but I'm unaware of any hook we could use to dump the logging buffer when that happens.